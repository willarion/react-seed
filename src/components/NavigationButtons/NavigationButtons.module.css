.buttons_center {
  margin: 10px auto 0;
}