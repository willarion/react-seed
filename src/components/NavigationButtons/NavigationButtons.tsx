import React from 'react';
import styles from './NavigationButtons.module.css';
import classNames from 'classnames';

interface NavigationButtonsProps {
  onForward: () => void;
  onBack: () => void;
  backButtonStatus: boolean;
  forwardButtonStatus: boolean;
}

export const NavigationButtons: React.FC<NavigationButtonsProps> = ({
  onForward,
  onBack,
  backButtonStatus,
  forwardButtonStatus,
}) => {
  return (
    <div className={classNames('controlButtons', styles.buttons_center)}>
      <button
        onClick={onBack}
        type="button"
        key="back-button"
        className="btn btn-default btn-prevNext"
        disabled={backButtonStatus}
      >
        <i className="icon-left-open-big" />
        Previous
      </button>
      <button
        onClick={onForward}
        type="button"
        key="forward-button"
        className="btn btn-default btn-prevNext"
        disabled={forwardButtonStatus}
      >
        Next
        <i className="icon-right-open-big" />
      </button>
    </div>
  );
};
