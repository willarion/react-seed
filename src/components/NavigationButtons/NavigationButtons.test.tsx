import { shallow } from 'enzyme';
import { NavigationButtons } from './NavigationButtons';

const mockFunction = jest.fn();

describe('component/NavigationButtons', () => {
  beforeEach(() => jest.clearAllMocks());

  it('should render with no disabled btn', () => {
    expect(
      shallow(
        <NavigationButtons
          onForward={mockFunction}
          onBack={mockFunction}
          onBackButton={false}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('should render with disabled btn', () => {
    expect(
      shallow(
        <NavigationButtons
          onForward={mockFunction}
          onBack={mockFunction}
          onBackButton={true}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('should call function on before btn click', () => {
    const wrapper = shallow(
      <NavigationButtons
        onForward={mockFunction}
        onBack={mockFunction}
        onBackButton={true}
      />,
    );
    const event = { data: 'click event' };
    const button = wrapper.findWhere((node) => node.key() === 'back-button');
    button.simulate('click', event);
    expect(mockFunction).toBeCalled();
  });

  it('should call function on forward btn click', () => {
    const wrapper = shallow(
      <NavigationButtons
        onForward={mockFunction}
        onBack={mockFunction}
        onBackButton={true}
      />,
    );
    const event = { data: 'click event' };
    const button = wrapper.findWhere((node) => node.key() === 'forward-button');
    button.simulate('click', event);
    expect(mockFunction).toBeCalled();
  });
});
