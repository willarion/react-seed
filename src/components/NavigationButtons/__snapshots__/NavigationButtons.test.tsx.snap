// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/NavigationButtons should render with disabled btn 1`] = `
<div
  className="controlButtons"
>
  <button
    className="btn btn-default btn-prevNext"
    key="back-button"
    onClick={[MockFunction]}
    type="button"
  >
    <i
      className="icon-left-open-big"
    />
    Previous
  </button>
  <button
    className="btn btn-default btn-prevNext"
    key="forward-button"
    onClick={[MockFunction]}
    type="button"
  >
    Next
    <i
      className="icon-right-open-big"
    />
  </button>
</div>
`;

exports[`component/NavigationButtons should render with no disabled btn 1`] = `
<div
  className="controlButtons"
>
  <button
    className="btn btn-default btn-prevNext"
    key="back-button"
    onClick={[MockFunction]}
    type="button"
  >
    <i
      className="icon-left-open-big"
    />
    Previous
  </button>
  <button
    className="btn btn-default btn-prevNext"
    key="forward-button"
    onClick={[MockFunction]}
    type="button"
  >
    Next
    <i
      className="icon-right-open-big"
    />
  </button>
</div>
`;
